src/NHibernate/Type/TimestampType.cs

```diff
@@ -1,142 +1,50 @@
 using System;
-using System.Collections;
-using System.Data.Common;
 using NHibernate.Engine;
-using NHibernate.SqlTypes;
 
 namespace NHibernate.Type
 {
 	/// <summary>
-	/// This is almost the exact same type as the DateTime except it can be used
-	/// in the version column, stores it to the accuracy the database supports, 
-	/// and will default to the value of DateTime.Now if the value is null.
+	/// This is almost the exact same type as the <see cref="DateTimeType" /> excepted it can be used
+	/// in the version column, stores it to the accuracy the database supports
+	/// and will default to the value of <see cref="TimestampType.Now" /> if the value is not valid.
```

Comment:
We may remove that since I do not think it was intended, the aim for the code being to replace `null` values, but in a method which is never called on `null`. It coud be a breaking change for edge cases, where a `timestamp` is mapped to an `object` property in which the app may put non `DateTime` data...

I have left the accuracy part in case for some reason we were to reduce `DateTimeType` one again.